python_modules/dagster/dagster/components/component/component.py

```diff
@@ -102,6 +102,14 @@ def get_additional_scope(cls) -> Mapping[str, Any]:
     @abstractmethod
     def build_defs(self, context: "ComponentLoadContext") -> Definitions: ...
 
+    def build_defs_and_cache(self, context: "ComponentLoadContext") -> Definitions:
+        from dagster.components.core.context import ComponentsLoadType
+
+        defs = self.build_defs(context)
+        if context.load_type == ComponentsLoadType.INITIAL_LOAD:
+            context.record_component_defs(defs)
```

Comment:
Stash the mapping of component -> produced defs to the context, which gets serialized into StateBackedDefinitionsLoader data